File: 2015/day01/day01.py
with open('day01.input') as f:
    s = f.read()
print(s.count('(')-s.count(')'))

floor = 0
for i, c in enumerate(s, 1):
    floor += {'(': 1, ')': -1}[c]
    if floor == -1:
        break
print(i)

File: 2015/day01/day01.nim
import strutils

const s = slurp "day01.input"

echo s.count('(') - s.count(')')

var floor: int
for i, c in s:
  floor += (if c == '(': 1 else: -1)
  if floor == -1:
    echo i+1
    break
